animation_nodes/algorithms/mesh_generation/find_shortest_path.pyx

```diff
@@ -0,0 +1,113 @@
+from ... data_structures import GPStroke
+from ... data_structures cimport (
+    Mesh,
+    LongList,
+    ColorList,
+    FloatList,
+    PolySpline,
+    DoubleList,
+    BooleanList,
+    Vector3DList,
+)
+
+from libc.math cimport INFINITY
+from . line import getLinesMesh
+from ... math cimport distanceVec3
+
+# Dijkstra's algorithm (https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm) but is implemented
+# such a way to handle multi-sources and mesh with multiple islands.
+def getShortestPath(Mesh mesh, LongList sources, Py_ssize_t target):
+    cdef LongList previousVertices = computeShortestPathTree(mesh, sources, target)
+    cdef LongList indices
+    cdef Py_ssize_t index
+
+    indices = LongList()
+    index = target
+    if previousVertices.data[index] == -1: return indices
+    while index != -1:
+        indices.append(index)
+        index = previousVertices.data[index]
+
+    return indices.reversed()
+
+def getShortestTree(Mesh mesh, LongList sources, str pathType):
+    cdef Vector3DList vertices = mesh.vertices
+    cdef long vertexCount = vertices.length
+    cdef LongList previousVertices = LongList(length = vertexCount)
+    previousVertices = computeShortestPathTree(mesh, sources)
+
+    cdef Vector3DList sortLocations
+    cdef Py_ssize_t i, index
+    cdef list tree = []
+
+    for i in range(vertexCount):
+        sortLocations = Vector3DList()
+        index = i
+        if previousVertices.data[index] == -1: continue
+        while index != -1:
+            sortLocations.append(vertices[index])
+            index = previousVertices.data[index]
+
+        tree.append(constructPath(pathType, sortLocations))
+
+    return tree
+
+def constructPath(str pathType, Vector3DList sortLocations):
+    cdef long amount
+    if pathType == "MESH":
+        return getLinesMesh(sortLocations.reversed(), False)
+    elif pathType == "SPLINE":
+        return PolySpline.__new__(PolySpline, sortLocations.reversed())
+    elif pathType == "STROKE":
+        amount = sortLocations.length
+        strengths = FloatList(length = amount)
+        pressures = FloatList(length = amount)
+        uvRotations = FloatList(length = amount)
+        vertexColors = ColorList(length = amount)
+        strengths.fill(1)
+        pressures.fill(1)
+        uvRotations.fill(0)
+        vertexColors.fill((0, 0, 0, 0))
```

Comment:
It will be faster if this was `fill(0)`.

Reply:
Done.